containerd-shim/process_solaris.go

```diff
@@ -0,0 +1,63 @@
+// +build solaris
+
+package main
+
+import (
+	"io"
+	"os"
+	"syscall"
+)
+
+// setPDeathSig is a no-op on Solaris as Pdeathsig is not defined.
+func setPDeathSig() *syscall.SysProcAttr {
+	return nil
+}
+
+// openIO opens the pre-created fifo's for use with the container
+// in RDWR so that they remain open if the other side stops listening
+func (p *process) openIO() error {
```

Comment:
I've not moved to this yet for the Solaris platform for two reasons: 1. The syscall package isn't supported on Solaris.fifos needs to call unix.Mkfifo instead of syscall.Mkfifo amongst other things for it to work on Solaris.
2. context.Background() is timing out. I need to figure out why thats happening before I can switch to that logic.
I've added a TODO for it. I'll switch to using fifos as soon as both of these are resolved.